test/AutoDiff/activity_analysis.swift

```diff
@@ -71,27 +71,27 @@ func testArrayUninitializedIntrinsic(_ x: Float, _ y: Float) -> [Float] {
 // CHECK: [NONE]   // function_ref _allocateUninitializedArray<A>(_:)
 // CHECK: [ACTIVE]   %6 = apply %5<Float>(%4) : $@convention(thin) <τ_0_0> (Builtin.Word) -> (@owned Array<τ_0_0>, Builtin.RawPointer)
 // CHECK: [ACTIVE] (**%7**, %8) = destructure_tuple %6 : $(Array<Float>, Builtin.RawPointer)
-// CHECK: [VARIED] (%7, **%8**) = destructure_tuple %6 : $(Array<Float>, Builtin.RawPointer)
-// CHECK: [VARIED]   %9 = pointer_to_address %8 : $Builtin.RawPointer to [strict] $*Float
-// CHECK: [VARIED]   %11 = integer_literal $Builtin.Word, 1
-// CHECK: [VARIED]   %12 = index_addr %9 : $*Float, %11 : $Builtin.Word
+// CHECK: [ACTIVE] (%7, **%8**) = destructure_tuple %6 : $(Array<Float>, Builtin.RawPointer)
+// CHECK: [ACTIVE]   %9 = pointer_to_address %8 : $Builtin.RawPointer to [strict] $*Float
+// CHECK: [ACTIVE]   %11 = integer_literal $Builtin.Word, 1
+// CHECK: [ACTIVE]   %12 = index_addr %9 : $*Float, %11 : $Builtin.Word
 
 @differentiable(where T: Differentiable)
 func testArrayUninitializedIntrinsicGeneric<T>(_ x: T, _ y: T) -> [T] {
   return [x, y]
 }
 
 // CHECK-LABEL: [AD] Activity info for ${{.*}}testArrayUninitializedIntrinsicGeneric{{.*}} at (source=0 parameters=(0 1))
-// CHECK: [VARIED] %0 = argument of bb0 : $*T
-// CHECK: [VARIED] %1 = argument of bb0 : $*T
+// CHECK: [ACTIVE] %0 = argument of bb0 : $*T
+// CHECK: [ACTIVE] %1 = argument of bb0 : $*T
 // CHECK: [USEFUL]   %4 = integer_literal $Builtin.Word, 2
 // CHECK: [NONE]   // function_ref _allocateUninitializedArray<A>(_:)
 // CHECK: [ACTIVE]   %6 = apply %5<T>(%4) : $@convention(thin) <τ_0_0> (Builtin.Word) -> (@owned Array<τ_0_0>, Builtin.RawPointer)
 // CHECK: [ACTIVE] (**%7**, %8) = destructure_tuple %6 : $(Array<T>, Builtin.RawPointer)
-// CHECK: [VARIED] (%7, **%8**) = destructure_tuple %6 : $(Array<T>, Builtin.RawPointer)
-// CHECK: [VARIED]   %9 = pointer_to_address %8 : $Builtin.RawPointer to [strict] $*T
-// CHECK: [VARIED]   %11 = integer_literal $Builtin.Word, 1
-// CHECK: [VARIED]   %12 = index_addr %9 : $*T, %11 : $Builtin.Word
+// CHECK: [ACTIVE] (%7, **%8**) = destructure_tuple %6 : $(Array<T>, Builtin.RawPointer)
```

Comment:
Note: with this patch, some non-differentiable values (e.g. `Builtin.RawPointer` values) get marked as active. In a later patch, I discovered that this triggers a crash pullback generation, which currently assumes that all active values are differentiable when propagating adjoint values of bb args to incoming values.

This seems like a reason to revisit https://github.com/apple/swift/pull/28259/files#r346257819: we could selectively propagate usefulness through `pointer_to_address` users' operands instead of doing a "blanket" propagation.

Reply:
> Note: with this patch, some non-differentiable values (e.g. `Builtin.RawPointer` values) get marked as active. In a later patch, I discovered that this triggers a crash during pullback generation, which currently assumes that all active values are differentiable when propagating adjoint values of bb args to incoming values.

I actually found that the crash (TF-966) exists both with and without this patch, so it's orthogonal. Regarding https://github.com/apple/swift/pull/28259#discussion_r346257819, I decided to selectively propagate usefulness to avoid unnecessarily marking values as active.